src/hotspot/share/runtime/handshake.cpp

```diff
@@ -627,12 +627,13 @@ void HandshakeState::do_self_suspend() {
 
 // This is the closure that prevents a suspended JavaThread from
 // escaping the suspend request.
-class ThreadSuspensionHandshake : public AsyncHandshakeClosure {
+class ThreadSelfSuspensionHandshake : public AsyncHandshakeClosure {
  public:
-  ThreadSuspensionHandshake() : AsyncHandshakeClosure("ThreadSuspension") {}
+  ThreadSelfSuspensionHandshake() : AsyncHandshakeClosure("ThreadSuspension") {}
   void do_thread(Thread* thr) {
-    JavaThread* target = thr->as_Java_thread();
-    target->handshake_state()->do_self_suspend();
+    JavaThread* thread_current = thr->as_Java_thread();
+    assert(thread_current == Thread::current(), "Must be self executed.");
+    thread_current->handshake_state()->do_self_suspend();
```

Comment:
nit - When you know it is the "current" thread, @dholmes-ora has been
using just `current` and not `current_thread` or `thread_current`.